### PR TITLE
[MBL-17949][All] Use common inbox

### PR DESCRIPTION
**Pull request by domonkosadam**
Test plan: Test the entire inbox in all three application.

refs: MBL-17949
affects: Student, Teacher
release note: Inbox improvements

## Checklist

- [ ] Run E2E test suite
- [x] Tested in dark mode
- [x] Tested in light mode
- [ ] Approve from product

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — CHANGES_REQUESTED** [review]
Only tested the Teacher app on tablet. Found some important issues:

- [ ] We should handle page views and screen views on the new inbox screens
- [ ] Teacher split view cases are not handled at all

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

File: apps/teacher/src/main/java/com/instructure/teacher/features/inbox/compose/TeacherInboxComposeRepository.kt [hunk 1]
```diff
@@ -15,56 +15,45 @@
...
-        attachments: List<Attachment>,
-        context: CanvasContext
-    ): DataResult<Conversation> {
-        TODO("Not yet implemented")
+        return DataResult.Success(validCourses)
     }
 
-    override suspend fun canSendToAll(context: CanvasContext): DataResult<Boolean> {
-        TODO("Not yet implemented")
+    override suspend fun getGroups(forceRefresh: Boolean): DataResult<List<Group>> {
```
[the remark this answers is not in the document]
Comment (domonkosadam) [reply apps/teacher/src/main/java/com/instructure/teacher/features/inbox/compose/TeacherInboxComposeRepository.kt:51]: I'm not sure about that either, but I checked the previous implementation and there was a network request to get the groups.

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

File: apps/teacher/src/main/java/com/instructure/teacher/features/inbox/compose/TeacherInboxComposeRepository.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (tamaskozmer) [reply apps/teacher/src/main/java/com/instructure/teacher/features/inbox/compose/TeacherInboxComposeRepository.kt:51]: If think we can leave this out, we don't request the groups in the Calendar either.

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

File: libs/pandautils/src/main/java/com/instructure/pandautils/features/inbox/compose/InboxComposeViewModel.kt [hunk 2]
```diff
@@ -351,6 +331,46 @@ class InboxComposeViewModel @Inject constructor(
         }
     }
 
+    fun cancelDismissDialog(isShow: Boolean) {
```
Comment (tamaskozmer) [inline libs/pandautils/src/main/java/com/instructure/pandautils/features/inbox/compose/InboxComposeViewModel.kt:334]: These new functions should be unit tested as well.

File: apps/teacher/src/main/java/com/instructure/teacher/activities/BottomSheetActivity.kt [hunk 3]
```diff
@@ -26,6 +26,7 @@ import android.content.Intent
 import android.graphics.Color
 import android.os.Bundle
 import android.os.Parcelable
+import android.util.Log
```
Comment (tamaskozmer) [inline apps/teacher/src/main/java/com/instructure/teacher/activities/BottomSheetActivity.kt:29]: This import is unused.

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

**Comment by kdeakinstructure** [comment]
QA Findings:


- If there is no attachment, the expand/collapse arrow icon makes no sense and can be confusing (see 1. video)

- 'All people in course' (or All people in teachers, etc..) counts the current user but when he/She navigates back then he/she won'T be in the list (so for example all people counts 6 people but when replying to that message the recipients will be shown one by one and the current user is not among them so only 5 people will be displayed (see 1. video)

- FAB and filter colors are different in student and teacher app (maybe it's just a setting? I don't really know what is the expected color here.... brand color, or course color, etc.) (see 2. video)

- In parent the replies look very different. I thought if the inbox will be common that it mean it should be really the exact same in all apps.  (see screenshot)

https://github.com/user-attachments/assets/1f36cc8b-cc84-452a-be38-4fc8111970ee

https://github.com/user-attachments/assets/e9f725b4-600b-494e-891d-c2c904acb48b

(I couldn't upload my screenshot taken in Logcat because GitHub says it's a "big" file, and I should try again with file size less than 10MB. This is weird because the file actually 604.55 kB....)

**Comment by domonkosadam** [comment]
1. I think that's the correct behaviour as only 1 line is shown in the collapsed state so longer messages can be read as well. It indeed makes no sense when the text message is only one line long but I think it's better to handle all messages universally.
2. Same as web. It's just a shortcut to handle recipients and with the current API users can't send messages to themselves.
3. I will check it, but that screen is common in the current version as well.
4. What do you mean by very different? Can you provide me some more screenshots?
5. It's just as design. I think it's ok because reply is the most common action.
6. Parent app can't handle pdf files so that's why I solved it like that. We could handle image and video attachments differently but that should be done in a different ticket.

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Pull request merged**